workflow/controller/operator.go

```diff
@@ -229,7 +229,11 @@ func (woc *wfOperationCtx) operate(ctx context.Context) {
 		woc.updated = wfUpdate
 		if !acquired {
 			woc.log.Warn("Workflow processing has been postponed due to concurrency limit")
-			woc.wf.Status.Message = msg
+			phase := woc.wf.Status.Phase
+			if phase == wfv1.WorkflowUnknown {
+				phase = wfv1.WorkflowPending
```

Comment:
so this ends up skipping the block on line 253. in particular, `createPDBResource` is skipped in this case. see https://github.com/argoproj/argo-workflows/issues/10178#issuecomment-2068596407.

we really shouldn't be creating resources for a not started Workflow without a lock though, so I'm not sure what the optimal way of fixing this is

Reply:
Filed an issue to follow-up on this as I don't have a solution off the top of my head: #12966